packages/graph/hash_graph/lib/graph/src/api/rest/mod.rs

```diff
@@ -40,6 +43,34 @@ fn api_documentation() -> Vec<openapi::OpenApi> {
     ]
 }
 
+async fn read_from_store<'pool, 'id: 'pool, T, O, P, I>(
```

Comment:
It's probably possible to introduce an `Identifier` trait to encapsulate these functions. Is this what you had in mind?

Reply:
I gave it another try and I wasn't able to reduce the amount of generic parameters. As mentioned above, it should somehow be possible to replace `O` with `<<<P as StorePool>::Store as Store<'pool>> as Read<'id, I, T>>::Output`, but this is (a) not readable and (b) requires some bounds on the Output which I'm not able to find out (`axum` implements their types (using macros 😒) on functions, and getting the bounds there correctly is a mess).

Theoretically, we only need one generic: `T`, but the other generics are used at least in two locations, so it's not possible to use `impl Trait` in argument positions here. It's only possible to use these function in all places because it is so generic 😄